### DISCUSSION TITLE
Content Block (with upload/multiple upload) inline creation

### DISCUSSION THREAD

**Discussion by manfield** [Laravel-Backpack/community-forum#1119] [Ideas]
I think this is would be a very useful feature. Almost necessary I guess.
If I had to use backpack as custom web content management system this feature would solve most of the requirements.
Content Blocks in a WCMS system are quite always made of title, rich text, media (or media gallery), attachments, video,...
This Content Block may be instantiated for a "page" or "post/article" multiple times and filled totally or partially. 
This way your page/post/article may be a sequence of mixed elements in a very versatile way.
Unfortunately this kind of inline creation does not work for upload or multiple uploads and content blocks are not added to the form directly but through a modal (though this kind of content should be entirely printed out inline).
I'd love to see this feature available. Probably Backpack would be the solution for many projects.

**Comment by jcastroa87** [comment]
Hello @manfield 

You think something like this package [File Manager](https://github.com/Laravel-Backpack/FileManager)? Or did you have something else in mind, please share an example, so maybe we can help you to achieve this.

Cheers.

**Comment by manfield** [comment]
Try imaging a repeating object like this, you can create n times (the object is a simplification that contains a multilple uploader to create an image gallery):

![Group 1](https://github.com/user-attachments/assets/0e06fd84-bb36-443d-ac09-b94bbb7aa1a4)

If I can repeat this object N times I can manage a lot of situation on a web page.
But it seems block repeating and multiple uploads don't work together.

**Reply by karandatwani92** [reply]
Hey @manfield 

You can [Create a custom Field Type](https://backpackforlaravel.com/docs/6.x/crud-fields#creating-a-custom-field-type-1) using [editorjs](https://editorjs.io/) for this custom need. I also wanted to introduce something like this, but It's not MIT-licensed. So, I canceled the plan for a while.
